### ISSUE TITLE
argparse drops "|" when a positional, then a special "-" argument (order-sensitive), are defined in a mutually exclusive group

### ISSUE THREAD

**Issue by keeeeymann** [python/cpython#113008]
# Bug report

### Bug description:

I'm defining a argparse parser that works like vim:

> support a file as a positional:
> ```shell
> vim file.py
> ```
> or reading from stdin:
> ```shell
> cat file.py | vim -
> ```

So I write a mutually exclusive group:
```python
import argparse
ap = argparse.ArgumentParser(description="fake vim")
mg = ap.add_mutually_exclusive_group(required=True)
mg.add_argument("input", nargs="?")
mg.add_argument("-", dest="from_stdin", action="store_true")
args = ap.parse_args()
```
When I use "-h" to show the help message, the "|" in usage line is dropped, as if the 2 arguments are not mutually exclusive:
```
usage: fake_vim.py [-h] [-] [input]
```
However, if I  ***swap*** the 2 `add_argument` lines in the source code:
```python
import argparse
ap = argparse.ArgumentParser(description="fake vim")
mg = ap.add_mutually_exclusive_group(required=True)
mg.add_argument("-", dest="from_stdin", action="store_true")
mg.add_argument("input", nargs="?")
args = ap.parse_args()
```
It magically behaves correctly:
```
usage: fake_vim.py [-h] (- | input)
```

### CPython versions tested on:

3.9, 3.11

### Operating systems tested on:

Windows

### Linked PRs
* gh-113085

**Comment by savannahostrowski** [comment]
I can repro with 3.9, 3.10 and 3.11.

**Comment by hpaulj** [comment]
two things

A mutually exclusive group can handle only one `positional` argument, and it should be the last one added to the group..  I don't recall if that's more a limitation in the testing or the usage.  Usage formatting is know to be brittle, especially with regard to  the MXGs.

A bare '-' dest is not encouraged.  It may be possible, but usually should something like '-f', dash plus single character.

Usage formatting needs a complete rewrite - one of these days.

**Comment by serhiy-storchaka** [comment]
Fixed in 3.14. Not sure if it's worth doing the backport -- it's only a minor improvement to the help output, and there's already a known workaround.

As for the original example, I do not recommend to use option "-" to solve this problem. It would work weird if you have more than one positional arguments. It is easier to simply check the positional argument value after parsing:
```py
args.from_stdin = (args.input == '-')
```
